.github/workflows/generate_pages.yml

```diff
@@ -0,0 +1,41 @@
+name: Deploy Yard to GitHub Pages
+
+on:
+  push:
+    branches:
+      - 'master'
+  workflow_dispatch:
+
+permissions:
```

Comment:
Did you look into whether this could be moved to within the `deploy` block?

Reply:
Yeah I did. But unfortunately it did not work. 

https://github.com/sanger/traction-service/actions/runs/10405310815

Reply:
Oh interesting!